benchexec/tools/ultimate.py

```diff
@@ -27,14 +27,52 @@ class UltimateTool(benchexec.tools.template.BaseTool):
     Abstract tool info for Ultimate-based tools.
     """
 
+    REQUIRED_PATHS = [
+              "artifacts.xml",
+              "config",
+              "configuration",
+              "data",
+              "features",
+              "p2",
+              "plugins",
+              "LICENSE",
+              "LICENSE.GPL",
+              "LICENSE.GPL.LESSER",
+              "README",
+              "Ultimate",
+              "Ultimate.ini",
+              "Ultimate.py",
+              "z3",
+              "mathsat",
+              "cvc4",
+              ]
+
     def executable(self):
         return util.find_executable('Ultimate.py')
 
     def version(self, executable):
         return self._version_from_tool(executable)
 
     def cmdline(self, executable, options, tasks, spec, rlimits):
-        return [executable] + [spec] + options + ['--full-output'] + tasks
+        if executable == None: 
+            raise Exception('No executable specified')
+        
+        cmdline = [executable] + ['--full-output']
+        
+        if spec != None: 
+            cmdline = cmdline + ['--spec'] + [spec]
+            
+        if tasks:
+            cmdline = cmdline + ['--file'] + tasks            
+            
+        if options:
+            for option in options:
+                if option in ['32bit','64bit']:
+                    cmdline = cmdline + ['--architecture'] + [option]
+                else:
+                    cmdline = cmdline + [option]
```

Comment:
This translates from the old 32bit / 64bit parameter to a named one for our wrapper script. I do not think this should change anything for users.